website/static/js/osfHelpers.js

```diff
@@ -179,7 +179,7 @@ var getAllNodeChildrenFromNodeList = function(parent, nodeList) {
         var node = remaining.pop();
         for (var c in tree[node]){
             var child = tree[node][c];
-            remaining.push(tree[child]);
+            remaining.push([child]);
```

Comment:
Since this is an isolated method and not part of a view model, this should be fairly easy to test. Could you add a basic test suite for this including this case? Single item, parent with child, parent with child and grandchild. If there's a couple of other variations that might be troublesome, go ahead and do those as well, but especially the current case. There's some good samples in fangorn.test.js.